source/Sashimi.Terraform.Tests/OnlyExposeWhatIsNecessary.ServerExtensionsShouldMinimiseWhatIsExposed.approved.txt

```diff
@@ -0,0 +1 @@
+Sashimi.Terraform.TerraformModule
```

Comment:
Instead of and approval file, I would prefer a typed refactor friendly class we can load by convention, eg:
```c#
// This class would be in the module
public class MyPublicsTest: OnlyExposeWhatIsNecessary
{
    MyPublicsTest()
    {
        Add<HereIsAPublicType>();
    }

}

// This nunit test would be in Calamari.Tests.Shared
[TestFixture]
public abstract class OnlyExposeWhatIsNecessary
{
    protected void Add<T>()
    {
        ...
    }

    [Test]
    public void ServerExtensionsShouldMinimiseWhatIsExposed()
    {
        //Scan for classes that impl Publics

        var allPublicThings = assembly.GetExportedTypes()
            .Select(t => t.FullName);

        // Compare and report on discrepancies
    }
}
```

Reply:
That's a great suggestion 👍